packages/yew-router/src/components/link.rs

```diff
@@ -9,16 +9,20 @@ use crate::{utils, Routable};
 
 /// Props for [`Link`]
 #[derive(Properties, Clone, PartialEq)]
-pub struct LinkProps<R, Q = ()>
+pub struct LinkProps<R, S = (), Q = ()>
```

Comment:
```suggestion
pub struct LinkProps<R, Q = (), S = ()>
```

I think it is better to add the state as the last generic so we can minimise the changes required to update.
Please also update the component definition so it has a consistent order.